include/model.php

```diff
@@ -748,53 +763,86 @@ public function filter_language_terms_orderby( $orderby, $args, $taxonomies ) {
 	}
 
 	/**
-	 * Adds the missing language terms for the secondary language taxonomies.
+	 * Maybe adds the missing language terms for 3rd party language taxonomies.
 	 *
 	 * @since 3.4
 	 *
 	 * @return void
 	 */
-	public function add_missing_secondary_language_terms() {
-		foreach ( $this->get_languages_list() as $language ) {
-			$this->update_secondary_language_terms( $language->slug, $language->name, $language );
+	public function maybe_add_missing_3rd_party_language_terms() {
+		$registered_taxonomies = array_diff(
+			$this->translatable_objects->get_taxonomy_names( array( 'language' ) ),
+			// Exclude the post and term language taxonomies from the list.
+			array( $this->post->get_tax_language(), $this->term->get_tax_language() )
+		);
+
+		if ( empty( $registered_taxonomies ) ) {
+			// No 3rd party language taxonomies.
+			return;
 		}
 
-		// Clear the cache, so the new `term_id` and `term_taxonomy_id` appear in the languages list.
-		$this->clean_languages_cache();
+		// We have at least one 3rd party language taxonomy.
+		$known_taxonomies = ! empty( $this->options['3rd_party_taxonomies'] ) && is_array( $this->options['3rd_party_taxonomies'] ) ? $this->options['3rd_party_taxonomies'] : array();
+		$new_taxonomies   = array_diff( $registered_taxonomies, $known_taxonomies );
+
+		if ( empty( $new_taxonomies ) ) {
+			// No new 3rd party language taxonomies.
+			return;
+		}
+
+		// We have at least one unknown 3rd party language taxonomy.
+		$this->add_missing_secondary_language_terms( $new_taxonomies );
+
+		// Keep the previous values, so this is triggered only once per taxonomy.
+		$this->options['3rd_party_taxonomies'] = array_merge( $known_taxonomies, $new_taxonomies );
+		update_option( 'polylang', $this->options );
 	}
 
 	/**
-	 * Returns the default language.
+	 * Adds the missing language terms for the secondary language taxonomies.
 	 *
 	 * @since 3.4
 	 *
-	 * @return PLL_Language|false Default language object, `false` if no language found.
+	 * @param string[]|null $taxonomies Optional. List of language taxonomies to deal with. An empty value means all of
+	 *                                  them. Defauls to all taxonomies.
+	 * @return void
+	 *
+	 * @phpstan-param array<non-empty-string>|null $taxonomies
 	 */
-	public function get_default_language() {
-		if ( empty( $this->options['default_lang'] ) ) {
-			return false;
+	public function add_missing_secondary_language_terms( array $taxonomies = null ) {
```

Comment:
I propose to rename to `create_language_terms()`. Is it useful for this method to be public? Is even this method useful now ? Would it be appropriate merge it to the calling method?

Reply:
Not needed anymore indeed.
Removed in [b8df0da](https://github.com/polylang/polylang/pull/1242/commits/b8df0da2d593306e720cac41ad7fb608a971c1c0).